src/test/java/dev/blaauwendraad/masker/json/ConcurrentMaskingTest.java

```diff
@@ -0,0 +1,85 @@
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class ConcurrentMaskingTest {
+    private static final Duration MAX_CONCURRENT_TEST_RUN = Duration.ofSeconds(10);
+    private static final String JSON_INPUT =
+            """
+        {
+            "userInfo": {
+                "personalInfo": {
+                    "firstName": "Breus",
+                    "lastName": "Blaauwendraad"
+                },
+                "ipAddress": "123.102.123",
+                "userId": 1234
+            }
+        }""";
+
+    private static final String EXPECTED_MASKED_OUTPUT =
+            """
+        {
+            "userInfo": {
+                "personalInfo": {
+                    "firstName": "***",
+                    "lastName": "***"
+                },
+                "ipAddress": "***",
+                "userId": 1234
+            }
+        }""";
+
+    @Test
+    void concurrentJsonPathMasking() throws ExecutionException, InterruptedException {
+        JsonMasker masker =
+                JsonMasker.getMasker(
+                        JsonMaskingConfig.builder()
+                                .maskJsonPaths("$.userInfo.personalInfo", "$.userInfo.ipAddress")
+                                .build());
+        maskConcurrently(Runtime.getRuntime().availableProcessors(), masker);
+    }
+
+    @Test
+    void concurrentKeyMasking() throws ExecutionException, InterruptedException {
+        JsonMasker masker =
+                JsonMasker.getMasker(
+                        JsonMaskingConfig.builder().maskKeys("firstName", "lastName", "ipAddress").build());
+        maskConcurrently(Runtime.getRuntime().availableProcessors(), masker);
+    }
+
+    private void maskConcurrently(int nThreads, JsonMasker masker) throws ExecutionException, InterruptedException {
+        ExecutorService executorService = Executors.newFixedThreadPool(nThreads);
+        List<Future<@Nullable String>> futures = new ArrayList<>();
+        for (int i = 0; i < nThreads; i++) {
+            Callable<@Nullable String> voidCallable =
+                    () -> {
+                        Instant testStartTime = Instant.now();
+                        String maskedOutput = null;
+                        while (Instant.now().isBefore(testStartTime.plus(MAX_CONCURRENT_TEST_RUN))) {
+                            maskedOutput = masker.mask(JSON_INPUT);
+                        }
+                        return maskedOutput;
+                    };
+            Future<String> submit = executorService.submit(voidCallable);
+            futures.add(submit);
+        }
+        for (Future<@Nullable String> future : futures) {
+            Assertions.assertThat(future.get()).isEqualTo(EXPECTED_MASKED_OUTPUT);
```

Comment:
Better to make this assert inside a loop, we expect that every `maskedOutput` is correct, not only the last one

Reply:
Also then it can be `Runnable` instead of `Callable` ;)